### ISSUE TITLE
can't pull from gcr.io

### ISSUE THREAD

**Issue by casualjim** [vmware/vic#5012]
**VIC version:**
master

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?

```
dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
datastore="${GOVC_DATASTORE}/${1-kov}"
publicnet="${PUBLIC_NETWORK-PublicNetwork}"
internalnet="${PRIVATE_NETWORK-InternalNetwork}"
fingerprint="$("$dir/get-fingerprint.sh" "$GOVC_URL" | cut -d '=' -f2)"

"$dir/../.bin/vic-machine-$(uname | tr '[:upper:]' '[:lower:]')" create \
  --name "${1-kov}" \
  --target "$GOVC_URL" \
  --thumbprint "$fingerprint" \
  --user "$GOVC_USERNAME" \
  --image-store "$datastore" \
  --password "$GOVC_PASSWORD" \
  --container-network "$publicnet" \
  --bridge-network "$internalnet" \
  --public-network "$publicnet" \
  --compute-resource "$GOVC_CLUSTER" \
  --timeout "10m" \
  --volume-store "${datastore}:default" \
  --tls-cname "kov-vch" \
  --debug 0 \
  --force
```

**Steps to reproduce:**

docker pull "gcr.io/google_containers/hyperkube:v1.6.2"

**Actual behavior:**

Pulling from google_containers/hyperkube
Error while pulling image manifest: name doesn't match what was requested, expected: google_containers/hyperkube, downloaded: unused

**Expected behavior:**

v1.6.2: Pulling from google_containers/hyperkube
6d827a3ef358: Pull complete 
a3ed95caeb02: Pull complete 
001d026ef0c3: Pull complete 
159671a680e7: Pull complete 
d0561d2a13c7: Pull complete 

**Logs:**

[logs.zip](https://github.com/vmware/vic/files/975382/logs.zip)

**Comment by rogeliosanchez** [comment]
@jzt @sflxn @cgtexmex CCing you guys on this one.  I was able to reproduce on my system as well.  

```
 ✘  ~/go/src/github.com/vmware/vic   master  docker -H 10.192.123.125:2376 --tls pull gcr.io/google_containers/hyperkube:v1.6.2
Pulling from google_containers/hyperkube
Error while pulling image manifest: name doesn't match what was requested, expected: google_containers/hyperkube, downloaded: unused
```

**Comment by jzt** [comment]
It looks like we expect `manifest.Name` to hold a meaningful value when decoding the manifest, and in this case, it doesn't:

```go
	if manifest.Name != options.Image {
		return nil, "", fmt.Errorf("name doesn't match what was requested, expected: %s, downloaded: %s", options.Image, manifest.Name)
	}
```
Digging deeper...

**Comment by anchal-agrawal** [comment]
Closing as dup of #4985.

**Comment by anchal-agrawal** [comment]
@casualjim , this has been fixed by #5101.
